kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/DefaultExpressionEvaluator.java

```diff
@@ -560,6 +577,15 @@ ColumnVector visitCoalesce(ScalarExpression coalesce) {
             );
         }
 
+        @Override
+        ColumnVector visitLike(final Predicate like) {
+            List<Expression> children = like.getChildren();
+            return LikeExpressionEvaluator.eval(
+                        children.stream()
+                        .filter(Objects::nonNull)
```

Comment:
do we need the `filter`? Same for above.

Reply:
without this, a null expression for the optional escape would results in `Expression null is not supported.` from `ExpressionVisitor.visit(...)`. 
of course one side effect of plain filtering is that a null expression for input(first) or pattern(pattern) would get removed and input expressions would be evaluated in a displaced context. 

what do you think?

Reply:
How about having just two child expressions when there is no escape in Predicate three child expressions when there is escape?

Reply:
yes, taking care of it in the transformation stage.